assets/aml-benchmark/components/src/utils/online_endpoint/aoai_online_endpoint.py

```diff
@@ -0,0 +1,224 @@
+from .online_endpoint import OnlineEndpoint, ResourceState
+from .online_endpoint_model import OnlineEndpointModel
+from utils.logging import get_logger
+
+
+logger = get_logger(__name__)
+
+
+class AOAIOnlineEndpoint(OnlineEndpoint):
+    """Class for AOAI online endpoint."""
+
+    def __init__(
+            self,
+            workspace_name: str,
+            resource_group: str,
+            subscription_id: str,
+            online_endpoint_model: OnlineEndpointModel,
+            online_endpoint_url: Optional[str] = None,
+            endpoint_name: Optional[str] = None,
+            deployment_name: Optional[str] = None,
+            sku: Optional[str] = None,
+            location: Optional[str] = None,
+            api_version: str = '2023-05-01',
+            connections_name: Optional[str] = None
+    ):
+        """Init method."""
+        # For AOAI endpoint, the account name is the same as the endpoint name.
+        super().__init__(
+            workspace_name,
+            resource_group,
+            subscription_id,
+            online_endpoint_url,
+            endpoint_name,
+            deployment_name,
+            sku,
+            online_endpoint_model,
+            connections_name
+        )
+        self._api_version = api_version
+        self._location = location
+        self._validate_settings()
+
+    @property
+    def scoring_url(self) -> str:
+        """Get the scoring url."""
+        if self._online_endpoint_url is None:
+            return self._get_scoring_url()
+        return self._online_endpoint_url
+
+    def get_endpoint_name_from_url(self) -> str:
+        """Get the endpoint."""
+        return self._online_endpoint_url.split('/')[2].split('.')[0]
+
+    def get_deployment_name_from_url(self) -> str:
+        """Get the deployment name."""
+        return self._online_endpoint_url.split('/')[5]
+
+    def endpoint_state(self) -> ResourceState:
+        """Check if the endpoint exists."""
+        resp = self._call_endpoint(get_requests_session().get, self._aoai_account_url)
+        logger.info("Calling {} returned {} with content {}.".format(
+            self._aoai_deployment_url, resp.status_code, self._get_content_from_response(resp)))
+        return self._get_resource_state(resp)
+
+    def deployment_state(self) -> ResourceState:
+        """Check if the deployment exists."""
+        resp = self._call_endpoint(get_requests_session().get, self._aoai_deployment_url)
+        logger.info("Calling {} returned {} with content {}.".format(
+            self._aoai_deployment_url, resp.status_code, self._get_content_from_response(resp)))
+        return self._get_resource_state(resp)
+
+    def create_endpoint(self):
+        """Create the endpoint."""
+        self._validate_model(validate_version=False)
+        payload = {
+            "location": self._location,
+            "kind": "OpenAI",
+            "sku": {
+                "name": 'S0'
```

Comment:
Is `S0` the only SKU for OpenAI? Can we confirm this?

Reply:
Let me check. I am trying to do it in UI and s0 is the only option for endpoint